File: Server/src/main/java/com/api/server/foodb/OntologySynonym.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "ontology_synonyms")
public class OntologySynonym {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ontology_term_id")
    private OntologyTerm ontologyTerm;

    @Size(max = 255)
    @Column(name = "synonym")
    private String synonym;

    @Size(max = 255)
    @Column(name = "external_id")
    private String externalId;

    @Size(max = 255)
    @Column(name = "external_srouce")
    private String externalSrouce;

    @Size(max = 255)
    @Column(name = "parent_id")
    private String parentId;

    @Size(max = 255)
    @Column(name = "parent_source")
    private String parentSource;

    @Size(max = 255)
    @Column(name = "comment")
    private String comment;

    @Size(max = 255)
    @Column(name = "curator")
    private String curator;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

}

File: DotNetServer/DotNetServer/Entities/OntologySynonym.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Entities;

[Table("ontology_synonyms")]
[Index("OntologyTermId", Name = "index_ontology_synonyms_on_ontology_term_id")]
public partial class OntologySynonym
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("ontology_term_id")]
    public int? OntologyTermId { get; set; }

    [Column("synonym")]
    [StringLength(255)]
    public string? Synonym { get; set; }

    [Column("external_id")]
    [StringLength(255)]
    public string? ExternalId { get; set; }

    [Column("external_srouce")]
    [StringLength(255)]
    public string? ExternalSrouce { get; set; }

    [Column("parent_id")]
    [StringLength(255)]
    public string? ParentId { get; set; }

    [Column("parent_source")]
    [StringLength(255)]
    public string? ParentSource { get; set; }

    [Column("comment")]
    [StringLength(255)]
    public string? Comment { get; set; }

    [Column("curator")]
    [StringLength(255)]
    public string? Curator { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }

    [ForeignKey("OntologyTermId")]
    [InverseProperty("OntologySynonyms")]
    public virtual OntologyTerm? OntologyTerm { get; set; }
}
